File: sdks/php/src/Models/EvaluationResult.php
<?php

declare(strict_types=1);

namespace PhaseFlag\Models;

/**
 * The result of evaluating a feature flag.
 *
 * Contains the resolved variation (if any) and a reason string that
 * indicates how the value was determined (e.g. "default",
 * "targeting_match", "percentage_rollout").
 */
final class EvaluationResult
{
    public function __construct(
        public readonly string $flagKey,
        public readonly ?string $variationId = null,
        public readonly ?string $variationKey = null,
        public readonly mixed $value = null,
        public readonly string $reason = 'default',
    ) {}

    /**
     * @return array<string, mixed>
     */
    public function toArray(): array
    {
        return [
            'flag_key'      => $this->flagKey,
            'variation_id'  => $this->variationId,
            'variation_key' => $this->variationKey,
            'value'         => $this->value,
            'reason'        => $this->reason,
        ];
    }

    /**
     * @param array<string, mixed> $data
     */
    public static function fromArray(array $data): static
    {
        return new static(
            flagKey: $data['flag_key'],
            variationId: $data['variation_id'] ?? null,
            variationKey: $data['variation_key'] ?? null,
            value: $data['value'] ?? null,
            reason: $data['reason'] ?? 'default',
        );
    }
}

File: sdks/dotnet/src/Models/EvaluationResult.cs
using System.Text.Json;
using System.Text.Json.Serialization;

namespace PhaseFlag.Models;

/// <summary>
/// The result of evaluating a feature flag, whether locally or via the remote API.
/// </summary>
public sealed class EvaluationResult
{
    /// <summary>The key of the flag that was evaluated.</summary>
    [JsonPropertyName("flag_key")]
    public string FlagKey { get; set; } = string.Empty;

    /// <summary>The identifier of the resolved variation, if any.</summary>
    [JsonPropertyName("variation_id")]
    public string? VariationId { get; set; }

    /// <summary>The key of the resolved variation, if any.</summary>
    [JsonPropertyName("variation_key")]
    public string? VariationKey { get; set; }

    /// <summary>
    /// The resolved value. This is a <see cref="JsonElement"/> representing the
    /// underlying JSON value (boolean, string, number, or object).
    /// </summary>
    [JsonPropertyName("value")]
    public JsonElement Value { get; set; }

    /// <summary>
    /// The reason the value was resolved: <c>default</c>, <c>targeting_match</c>,
    /// <c>percentage_rollout</c>, or <c>error</c>.
    /// </summary>
    [JsonPropertyName("reason")]
    public string Reason { get; set; } = "default";
}
